py-polars/polars/lazyframe/frame.py

```diff
@@ -1366,8 +1367,18 @@ def sort(
             )
 
         by = parse_into_list_of_expressions(by, *more_by)
-        descending = extend_bool(descending, len(by), "descending", "by")
-        nulls_last = extend_bool(nulls_last, len(by), "nulls_last", "by")
+        descending = extend_bool(descending, len(by), "descending", "by", condense=True)
+        nulls_last = extend_bool(nulls_last, len(by), "nulls_last", "by", condense=True)
+
+        if (len(descending) > 1 or len(nulls_last) > 1) and any(
```

Comment:
We need to check this when we convert the DSL to the IR.

Reply:
Done✌️